### DISCUSSION TITLE
Can I deploy a waypoint with high-availability? (i.e. multiple replicas)

### DISCUSSION THREAD

**Discussion by jessebye** [istio/istio#53825] [Q&A] [answered]
If we are using waypoints in production, we need to have at least 2 replicas to ensure HA. If the node that is running the waypoint dies, we don't want to experience an outage between services that used the waypoint. Is there a way to deploy a waypoint with multiple replicas, or even better, with autoscaling based on load?

**Comment by howardjohn** [comment, marked as answer]
You can attach an HPA to the deployment: https://istio.io/latest/docs/tasks/traffic-management/ingress/gateway-api/#resource-attachment-and-scaling (we should document this better!)
